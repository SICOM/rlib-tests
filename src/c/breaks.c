/*
 *  Copyright (C) 2016 SICOM Systems, INC.
 *
 *  Author: Zoltán Böszörményi <zboszormenyi@sicom.com>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of version 2 of the GNU General Public
 * License as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program; if not, write to the
 * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

#include <config.h>

#include <stdio.h>
#include <rlib.h>

char *data[9][5] = {
	[0][0] = "company",
	[0][1] = "region",
	[0][2] = "district",
	[0][3] = "store",
	[0][4] = "sales",

	[1][0] = "Foo, Inc.",
	[1][1] = "Region A",
	[1][2] = "District X",
	[1][3] = "Store #1",
	[1][4] = "1500.00",

	[2][0] = "Foo, Inc.",
	[2][1] = "Region A",
	[2][2] = "District X",
	[2][3] = "Store #2",
	[2][4] = "1300.00",

	[3][0] = "Foo, Inc.",
	[3][1] = "Region A",
	[3][2] = "District Y",
	[3][3] = "Store #3",
	[3][4] = "2000.00",

	[4][0] = "Foo, Inc.",
	[4][1] = "Region A",
	[4][2] = "District Y",
	[4][3] = "Store #4",
	[4][4] = "1800.00",

	[5][0] = "Foo, Inc.",
	[5][1] = "Region B",
	[5][2] = "District M",
	[5][3] = "Store #11",
	[5][4] = "2500.00",

	[6][0] = "Foo, Inc.",
	[6][1] = "Region B",
	[6][2] = "District M",
	[6][3] = "Store #12",
	[6][4] = "2300.00",

	[7][0] = "Foo, Inc.",
	[7][1] = "Region B",
	[7][2] = "District N",
	[7][3] = "Store #13",
	[7][4] = "3000.00",

	[8][0] = "Foo, Inc.",
	[8][1] = "Region B",
	[8][2] = "District N",
	[8][3] = "Store #14",
	[8][4] = "2800.00",
};

int main(int argc, char **argv) {
	rlib *r;

	if (argc == 1) {
		printf("usage: %s [ pdf | xml | txt | csv | html ]\n", argv[0]);
		return 1;
	}

	r = rlib_init();
	rlib_add_datasource_array(r, "local_array");
	rlib_add_query_array_as(r, "local_array", data, 9, 5, "data");
	rlib_add_report(r, "breaks.xml");
	rlib_set_output_format_from_text(r, argv[1]);
	rlib_set_output_parameter(r, "debugging", "yes");
	rlib_set_output_parameter(r, "only_quote_strings", "yes");
	rlib_set_locale(r, "en_US");
	rlib_execute(r);
	rlib_spool(r);
	rlib_free(r);

	return 0;
}
