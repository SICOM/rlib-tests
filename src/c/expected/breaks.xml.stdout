<?xml version="1.0" encoding="UTF-8" ?>
<rlib>
<part>
<part_page_footer>
<output>
</output>
</part_page_footer>
<part_header>
<output>
</output>
</part_header>
<part_page_header>
<output>
<line>
<data col="0" width="40" font_point="11" bold="1" italics="0" align="left" type="string" >Breaks Example                          </data>
</line>
</output>
</part_page_header>
<table>
<tr>
<td width="100.000000" ><report_header>
</report_header>
<field_headers>
<output>
<line>
<data col="1" width="7" font_point="9" bold="0" italics="0" align="left" >Company</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="2" width="8" font_point="9" bold="0" italics="0" align="left" >District</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="3" width="6" font_point="9" bold="0" italics="0" align="left" >Region</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="4" width="5" font_point="9" bold="0" italics="0" align="left" >Store</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="47" font_point="9" bold="0" italics="0" align="right" >                                          Sales</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_headers>
<break_header name="companybreak">
<output>
<line>
<data col="1" width="11" font_point="9" bold="0" italics="0" align="left" type="string" >Foo, Inc.  </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $1500.00</data>
</line>
</output>
</break_header>
<break_header name="regionbreak">
<output>
<line>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="2" width="11" font_point="9" bold="0" italics="0" align="left" type="string" >Region A   </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $1500.00</data>
</line>
</output>
</break_header>
<break_header name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District X</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $1500.00</data>
</line>
</output>
</break_header>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #1  </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $1500.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #2  </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $1300.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<break_footer name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District X</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $2800.00</data>
</line>
</output>
</break_footer>
<break_header name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District Y</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $2000.00</data>
</line>
</output>
</break_header>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #3  </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $2000.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #4  </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $1800.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<break_footer name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District Y</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $3800.00</data>
</line>
</output>
</break_footer>
<break_footer name="regionbreak">
<output>
<line>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="2" width="11" font_point="9" bold="0" italics="0" align="left" type="string" >Region A   </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $6600.00</data>
</line>
</output>
</break_footer>
<break_header name="regionbreak">
<output>
<line>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="2" width="11" font_point="9" bold="0" italics="0" align="left" type="string" >Region B   </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $2500.00</data>
</line>
</output>
</break_header>
<break_header name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District M</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $2500.00</data>
</line>
</output>
</break_header>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #11 </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $2500.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #12 </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $2300.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<break_footer name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District M</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $4800.00</data>
</line>
</output>
</break_footer>
<break_header name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District N</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $3000.00</data>
</line>
</output>
</break_header>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #13 </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $3000.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<field_details>
<output>
<line>
<data col="0" width="3" font_point="9" bold="0" italics="0" align="left" >   </data>
<data col="4" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >Store #14 </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="33" font_point="9" bold="0" italics="0" align="right" type="number" >                         $2800.00</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
</line>
</output>
</field_details>
<break_footer name="districtbreak">
<output>
<line>
<data col="0" width="2" font_point="9" bold="0" italics="0" align="left" >  </data>
<data col="3" width="10" font_point="9" bold="0" italics="0" align="left" type="string" >District N</data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                           $5800.00</data>
</line>
</output>
</break_footer>
<break_footer name="regionbreak">
<output>
<line>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="2" width="11" font_point="9" bold="0" italics="0" align="left" type="string" >Region B   </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                          $10600.00</data>
</line>
</output>
</break_footer>
<break_footer name="companybreak">
<output>
<line>
<data col="1" width="12" font_point="9" bold="0" italics="0" align="left" type="string" >Foo, Inc.   </data>
<data col="0" width="1" font_point="9" bold="0" italics="0" align="left" > </data>
<data col="5" width="35" font_point="9" bold="0" italics="0" align="right" type="number" >                          $17200.00</data>
</line>
</output>
</break_footer>
<report_footer>
<output>
</output>
</report_footer>
</td>
</tr>
</table>
</part>
</rlib>
